### PR TITLE
feat: User Authentication

### PR DESCRIPTION
**Pull request by CeoFred**
- added users authentication using email & password

### CODE REVIEW

**Comment by CeoFred** [comment]
This fixes #2

File: internal/handlers/auth.go [hunk 1]
```diff
@@ -63,6 +65,69 @@ var (
...
+		return helpers.Dispatch400Error(c, "email and password does not match", nil)
+	}
+	jwtToken, err := helpers.GenerateToken(constant.JWTSecretKey, user.Email, user.FirstName+" "+user.LastName)
+
+	if err != nil {
+		return helpers.Dispatch400Error(c, "something went wrong", err)
+	}
+
+	// check if account is active
+	if user.AccountStatus == int(Suspended) {
```
Comment (opensaucerer) [inline internal/handlers/auth.go]: @CeoFred Just wondering why the `AccountStatus` struct isn't of type `AccountStatus` for direct comparison as opposed to `int` wrapping...direct comparison enforces consistency.

File: internal/helpers/functions.go [hunk 2]
```diff
@@ -62,6 +62,10 @@ func HashPassword(password string) (string, error) {
 	return string(bytes), err
 }
 
+func CompareHash(hash, plain string) {
```
Comment (opensaucerer) [inline internal/helpers/functions.go]: @CeoFred This seems to be empty

**Review by opensaucerer — COMMENTED** [review]
Left a few comments

File: internal/handlers/auth.go (hunk 1 above)
Comment (CeoFred) [reply internal/handlers/auth.go]: `user.AccountStatus` comes back from the db as `int` and not `AccountStatus` type which is still an int type though. @opensaucerer

Comment (opensaucerer) [reply internal/handlers/auth.go]: @CeoFred Yeah...however, if the type specified on the struct is `AccountStatus` which is mapped against an `int` type as I think it currently is, then the db formatted should format it against `AccountStatus` without any issue...else, one could also intercept the formatter for `AccountStatus` quite easily. I think gorm exposes a serializer interface for this....but ideally, it shouldn't be needed as the type `AccountStatus` here is a simple type mapped against the `int` primitve, and `gorm` should just be able to serialize into it easily.

**Pull request merged**